sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Blobs/src/Config/BlobsExtensionConfigProvider.cs

```diff
@@ -110,6 +130,20 @@ private void InitializeBlobTriggerBindings(ExtensionConfigContext context)
             return GetContainer(blobAttribute);
         }
 
+        public async Task<HttpResponseMessage> ConvertAsync(HttpRequestMessage input, CancellationToken cancellationToken)
+        {
+            var functionName = HttpUtility.ParseQueryString(input.RequestUri.Query)["functionName"];
+            if (_blobTriggerQueueWriter == null)
+            {
+                _blobTriggerQueueWriter = await _blobTriggerQueueWriterFactory.CreateAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
same here, is it fine if double init?

Reply:
I added code to ensure `_blobTriggerQueueWriter`  is initialized only once using `SlimSemaphore`.  Simple `lock` does not work as we have an `async` call and `Lazy<>` does not work as we need to pass `cancellationtoken `